plugins/router/mikrotik_system

```diff
@@ -0,0 +1,488 @@
+#!/bin/bash
+#%# family=auto
+
+: << EOF
+=head1 NAME
+mikrotik_system - This plugin fetches multiple Values from a mikrotik device.
+
+=head1 CONFIGURATION
+tested with a RB750GR3, RB5009 and CRS309.
+Dependencies:
+- bc
+- sshpass
+
+A User is needed for this plugin to work:
+/user add name=munin group=read password=hallowelt address=<munin-server-ip>
+
+plugin config:
+[mt_system_<name>]
+user root
+env.ssh_user munin
+env.ssh_password hallowelt
+env.ssh_host 192.168.2.1
+
+=head1 AUTHOR
+=over 4
+=item * Michael Grote
+=back
+
+=head1 LICENSE
+Unknown license
+
+=head1 MAGIC MARKERS
+=begin comment
+These magic markers are used by munin-node-configure when installing
+munin-node.
+=end comment
+  #%# family=auto
+  #%# capabilities=autoconf suggest
+=cut
+EOF
+
+# setze Variablen mit default-Werten
+ssh_user=${ssh_user:-user}
+ssh_password=${ssh_password:-password}
+ssh_host=${ssh_host:-192.168.2.1}
+c=0 # zähler; wird für verschiedene Schleifen benötigt
+
+# Funktionen
+function get_name {
+  while read -r line; do
+    if echo "$line" | grep -q 'name:'; then
+      name=$(echo "$line" | grep name: | awk '{ print $2 }')
+    fi
+  done <<< "$data"
+}
+function get_ros_version {
+  while read -r line; do
+    if echo "$line" | grep -q 'version:'; then
+      if echo "$line" | grep version: | grep "7." > /dev/null; then
```

Comment:
regarding the final `grep` call:
* you should either escape the dot or add the option `-F` - otherwise the dot is interpreted as "any character"
* instead of redirecting the output, you can simply use `grep -q` - it has the same effect

Reply:
In my opinion `grep` uses regex only with these parameters:
```
   Pattern Syntax
       -E, --extended-regexp
              Interpret PATTERNS as extended regular expressions (EREs, see below).

       -F, --fixed-strings
              Interpret PATTERNS as fixed strings, not regular expressions.

       -G, --basic-regexp
              Interpret PATTERNS as basic regular expressions (BREs, see below).  This is the default.

       -P, --perl-regexp
              Interpret  PATTERNS  as  Perl-compatible regular expressions (PCREs).  This option is experimental when
              combined with the -z (--null-data) option, and grep -P may warn of unimplemented features.
```

But you are right, at the moment it matches also `1.7.3`...

Reply:
i would change it to `grep -E "^7\."`

Reply:
> In my opinion grep uses regex only with these parameters: [..]

It really uses regular expressions *everywhere*. But you can disabled these via `-F` (`--fixed-strings`).

Your approach of `grep -E "^7\."` - and you could even omit the `-E`, since the "extend expressions" mainly affect the interpretation of braces and pipes:
* basic: `(|)` are simple characters
* extended: `(|)` are used for grouping regular expressions

Anyway - your approach is fine, too, of course ...

Reply:
you learn something new every day... it is changed